ENGAGEHF/Managers/MessageManager/MessageManager.swift

```diff
@@ -63,17 +67,74 @@ final class MessageManager: Manager {
         if let account, account.signedIn {
             updateSnapshotListener(for: account.details)
         }
+        
+        startProcessingCleanupTimer()
+    }
+    
+    private func startProcessingCleanupTimer() {
+        activeProcessingTimer?.cancel()
+        activeProcessingTimer = Task { @MainActor in
+            while !Task.isCancelled {
+                cleanupExpiredProcessingStates()
+                try? await Task.sleep(nanoseconds: 1_000_000_000) // Check every second
+            }
+        }
     }
     
+    @MainActor
+    private func cleanupExpiredProcessingStates() {
+        processingStates = processingStates.filter { $0.value.isStillProcessing }
+        
+        // Update messages that might have expired processing states
+        for index in messages.indices {
+            if let state = messages[index].processingState, !state.isStillProcessing {
+                messages[index].processingState = nil
+            }
+        }
+    }
     
+    @MainActor
+    func markAsProcessing(type: ProcessingState.ProcessingType) {
+        let correlationId = UUID().uuidString
+        let state = ProcessingState(
+            startTime: Date(),
+            type: type,
+            correlationId: correlationId
+        )
+        
+        processingStates[correlationId] = state
+        
+        // Update all related messages' processing states
+        for index in messages.indices where messages[index].isRelatedTo(state) {
+            messages[index].processingState = state
+        }
+        
+        // Schedule state cleanup
+        Task {
+            try? await Task.sleep(nanoseconds: 60_000_000_000) // 60 seconds
```

Comment:
Same here: See: https://developer.apple.com/documentation/swift/task/sleep(for:tolerance:clock:) - I hope I added it in all places, please check if there are other usages of `Task.sleep(nanoseconds:)` to make this consistent. Of course only talking about your own changes.